src/pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -513,63 +546,76 @@ def __init__(self, extra_options):
                 )
 
         # Options not yet compatible with contact resistance
-        if options["contact resistance"] == "true":
-            if options["operating mode"] == "explicit power":
+        _ensure_tuple(options["contact resistance"])
+        _ensure_tuple(options["operating mode"])
+
+        if "true" in options["contact resistance"]:
+            if "explicit power" in options["operating mode"]:
                 raise NotImplementedError(
                     "Contact resistance not yet supported for explicit power."
                 )
-            if options["operating mode"] == "explicit resistance":
+            if "explicit resistance" in options["operating mode"]:
                 raise NotImplementedError(
                     "Contact resistance not yet supported for explicit resistance."
                 )
 
         # Options not yet compatible with particle-size distributions
-        if options["particle size"] == "distribution":
-            if options["lithium plating porosity change"] != "false":
+
+        # Ensure all necessary options are tuples
+        for option in [
+            "particle size",
+            "lithium plating porosity change",
+            "heat of mixing",
+            "particle",
+            "particle mechanics",
+            "particle shape",
+            "SEI",
+            "stress-induced diffusion",
+            "thermal",
+        ]:
+            _ensure_tuple(options[option])
```

Comment:
```suggestion
        for _ in [
            "particle size",
            "lithium plating porosity change",
            "heat of mixing",
            "particle",
            "particle mechanics",
            "particle shape",
            "SEI",
            "stress-induced diffusion",
            "thermal",
        ]:
            _ensure_tuple(options[_])
```

As a convention, for variables that are not needed later in the code and are used only during the context of loops, we can use `_` to define them.